test-system/end-to-end.js

```diff
@@ -231,62 +240,102 @@ contract("End to end Colony network and Reputation mining testing", function(acc
         }
       ];
 
-      colonyTaskProps.forEach(async taskProp => {
-        const { colony } = colonies[taskProp.colonyIdx];
-        await setupFinalizedTask({
-          colonyNetwork,
-          colony,
-          domainId: taskProp.domainId,
-          skillId: taskProp.skillId,
-          manager: MANAGER,
-          evaluator: EVALUATOR,
-          worker: WORKER,
-          managerPayout: taskProp.managerPayout,
-          evaluatorPayout: taskProp.evaluatorPayout,
-          workerPayout: taskProp.workerPayout,
-          managerRating: taskProp.managerRating,
-          workerRating: taskProp.workerRating
-        });
-      });
+      await Promise.all(
+        colonyTaskProps.map(async taskProp => {
+          const { colony } = colonies[taskProp.colonyIdx];
+          await setupFinalizedTask({
+            colonyNetwork,
+            colony,
+            domainId: taskProp.domainId,
+            skillId: taskProp.skillId,
```

Comment:
Wait, isn't this `undefined` now?

Reply:
Sure was. But the helpers set default values, so it all still worked.